components/script/script_thread.rs

```diff
@@ -1471,28 +1476,310 @@ impl ScriptThread {
+        // as well as those for which a rendering update would be unnecessary,
+        // but this isn't happening here.
+        let pipeline_and_docs: Vec<(PipelineId, DomRoot<Document>)> = self
+            .documents
+            .borrow()
+            .iter()
+            .map(|(id, document)| (id, DomRoot::from_ref(&*document)))
+            .collect();
+        // Note: the spec reads: "for doc in docs" at each step
+        // whereas this runs all steps per doc in docs.
+        for (pipeline_id, document) in pipeline_and_docs {
+            // TODO(#32004): The rendering should be updated according parent and shadow root order
+            // in the specification, but this isn't happening yet.
+
+            // TODO(#31581): The steps in the "Revealing the document" section need to be implemente
+            // `process_pending_compositor_events` handles the focusing steps as well as other events
+            // from the compositor.
+
+            // TODO: Should this be broken and to match the specification more closely? For instance see
+            // https://html.spec.whatwg.org/multipage/#flush-autofocus-candidates.
+            self.process_pending_compositor_events(pipeline_id);
+
+            // TODO(#31665): Implement the "run the scroll steps" from
+            // https://drafts.csswg.org/cssom-view/#document-run-the-scroll-steps.
+
+            for (size, size_type) in document.window().steal_resize_events().into_iter() {
+                // Resize steps.
+                self.run_the_resize_steps(pipeline_id, size, size_type);
+
+                // Evaluate media queries and report changes.
+                document
+                    .window()
+                    .evaluate_media_queries_and_report_changes();
+
+                // https://html.spec.whatwg.org/multipage/#img-environment-changes
+                // As per the spec, this can be run at any time.
+                document.react_to_environment_changes()
             }
+
+            // Update animations and send events.
+            self.update_animations_and_send_events();
+
+            // TODO(#31866): Implement "run the fullscreen steps" from
+            // https://fullscreen.spec.whatwg.org/multipage/#run-the-fullscreen-steps.
+
+            // TODO(#31868): Implement the "context lost steps" from
+            // https://html.spec.whatwg.org/multipage/#context-lost-steps.
+
+            // Run the animation frame callbacks.
+            document.tick_all_animations();
+
+            // TODO(#31006): Implement the resize observer steps.
+
+            // TODO(#31870): Implement step 17: if the focused area of doc is not a focusable area,
+            // then run the focusing steps for document's viewport.
+
+            // TODO: Perform pending transition operations from
+            // https://drafts.csswg.org/css-view-transitions/#perform-pending-transition-operations.
+
+            // TODO(#31021: Run the update intersection observations steps from
```

Comment:
Nit. This is probably my fault, sorry. I'll fix this one.

```suggestion
            // TODO(#31021): Run the update intersection observations steps from
```